CHAID/CHAID.py

```diff
@@ -248,35 +268,57 @@ def generate_best_split(self, ind, dep):
             mappings = MappingDict()
             frequencies = {}
             for col in unique:
-                counts = np.unique(dep[index == col][0], return_counts=True)
-                frequencies[col] = cl.defaultdict(int)
-                frequencies[col].update(np.transpose(counts))
+                if self.d_v_type == CHAIDVariableTypes(2):
+                    frequencies[col] = dep[index == col]
+                else:
+                    counts = np.unique(dep[index == col][0], return_counts=True)
+                    frequencies[col] = cl.defaultdict(int)
+                    frequencies[col].update(np.transpose(counts))
 
             while len(unique) > 1:
                 size = int((len(unique) * (len(unique) - 1)) / 2)
                 sub_data = np.ndarray(shape=(size, 3), dtype=object, order='F')
-                for j, comb in enumerate(it.combinations(unique, 2)):
+
+                if self.i_v_types[i] == CHAIDVariableTypes(0):
+                    combinations = it.combinations(unique, 2)
+                else:
+                    combinations = list(it.combinations(unique, 2))
```

Comment:
` for k in unique: combinations += [(min(mappings[k]) - 1, max(mappings[k])), (min(mappings[k]), max(mappings[k])+1)]`